flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterClient.java

```diff
@@ -211,18 +185,45 @@ public void run() {
 		Runtime.getRuntime().addShutdownHook(clientShutdownHook);
 
 		isConnected = true;
+
+		logAndSysout("Waiting until all TaskManagers have connected");
+
+		while(true) {
```

Comment:
Yes, I wanted to bring up the Yarn session only once the cluster is ready. It is a semantic change but IMHO transparent to the user. There is not disadvantage from waiting until the cluster is ready. Ultimately, it would be nice to get rid of all waiting but we're not quite there yet.